lib/mqtt_abstraction_impl.cpp

```diff
@@ -486,6 +486,12 @@ bool MQTTAbstractionImpl::connectBroker(std::string& socket_path) {
     if (mqtt_connect(&this->mqtt_client, nullptr, nullptr, nullptr, 0, nullptr, nullptr, connect_flags,
                      mqtt_keep_alive) == MQTT_OK) {
         // TODO(kai): async?
+        MQTTErrors error = mqtt_sync(&this->mqtt_client);
+        if (error != MQTT_OK) {
+            EVLOG_error << fmt::format("Error during MQTT sync: {}", mqtt_error_str(error));
+            return false;
+        }
+
```

Comment:
The original code wasn't using early returns, but here might be a good place to change that.  Something like:
```
if (mqtt_connect(..) != MQTT_OK) {
  return false;
}

if ((const auto error = mqtt_sync(...)) != MQTT_OK) {
  EVLOG...;
  return false;
}

on_mqtt_connect();
return true
```

More readable imho.

As was mentioned in the comments already, a possible better solution would be to start the _event loop_/_the polling thread_ directly after doing the connect - because why would you create a socket and then not use it for some time?  It is indeed interesting to see that the mqttc implementation doesn't send anything on the `mqtt_connect` command - probably due to easier separation for error handling.  Good find @FaHaGit